plugins/events-backend-module-sqs/src/service/config.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Config } from '@backstage/config';
+
+const DEFAULT_WAIT_TIME_SECONDS_AFTER_EMPTY_RECEIVE = 60;
+const MAX_WAIT_SECONDS = 20;
+
+export interface AwsSqsEventSourceConfig {
+  pollingWaitTimeSeconds: number;
+  queueUrl: string;
+  region: string;
+  timeoutSeconds: number;
+  topic: string;
+  visibilityTimeoutSeconds?: number;
+  waitTimeSecondsAfterEmptyReceive: number;
+}
+
+export function readConfig(config: Config): AwsSqsEventSourceConfig[] {
+  const key = 'events.modules.awsSqs.topics';
+  const topics = config.getOptionalConfig(key);
+
+  return (
+    topics?.keys()?.map(topic => {
+      const topicConfig = topics.getConfig(topic);
+      const keyPrefix = `${key}.${topic}.`;
+
+      // queue config:
+      const pollingWaitTimeSeconds =
+        topicConfig.getOptionalNumber('queue.waitTimeSeconds') ??
+        MAX_WAIT_SECONDS;
+      if (
+        pollingWaitTimeSeconds < 0 ||
+        pollingWaitTimeSeconds > MAX_WAIT_SECONDS
+      ) {
+        throw new Error(
+          `${keyPrefix}queue.waitTimeSeconds must be within 0..20.`,
```

Comment:
missed opportunity to insert MAX_WAIT_SECONDS :)

Reply:
yes!